File: examples/JvThumbnail/BCB/JvThumbnailChildFormU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "JvThumbnailChildFormU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "JvBaseThumbnail"
#pragma link "JvExExtCtrls"
#pragma link "JvThumbImage"
#pragma link "JvThumbnails"
#pragma resource "*.dfm"
//TJvThumbnailChildForm *JvThumbnailChildForm;
//---------------------------------------------------------------------------
__fastcall TJvThumbnailChildForm::TJvThumbnailChildForm(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::FormShow(TObject *Sender)
{
  //thumbimage1.Picture.Free;
  titlePlaceGr->ItemIndex = ThumbNail1->TitlePlacement;
  AngleGr->ItemIndex      = ThumbImage1->Angle;
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::FileListBox1Change(TObject *Sender){
  if( FileListBox1->FileName!="")
  {
    ThumbNail1->FileName = FileListBox1->FileName;
  }
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::Button2Click(TObject *Sender)
{
  ThumbImage1->ChangeRGB(RedBar->Position,GreenBar->Position,BlueBar->Position);
  ThumbImage1->Contrast(contrastBar->Position);
  ThumbImage1->Lightness(LightnessBar->Position);
  RedBar->Position   = 0;
  GreenBar->Position = 0;
  BlueBar->Position   = 0;
  contrastBar->Position  = 0;
  LightnessBar->Position = 0;
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::ThumbNail1Click(TObject *Sender)
{
  if( ThumbNail1->FileName!="")
  {
    ThumbImage1->LoadFromFile(ThumbNail1->FileName);
  }
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::Button4Click(TObject *Sender)
{
  ThumbImage1->Invert();
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::Button5Click(TObject *Sender)
{
  ThumbImage1->Grayscale();
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::AngleGrClick(TObject *Sender)
{
    ThumbImage1->Angle = static_cast<TAngle >(AngleGr->ItemIndex);
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::asButtonCBClick(TObject *Sender)
{
  ThumbNail1->AsButton = asButtonCB->Checked;
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::autoloadCBClick(TObject *Sender)
{
  ThumbNail1->AutoLoad = autoloadCB->Checked;        
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::minMemCBClick(TObject *Sender)
{
  ThumbNail1->MinimizeMemory=minMemCB->Checked;
}
//---------------------------------------------------------------------------
void __fastcall TJvThumbnailChildForm::titlePlaceGrClick(TObject *Sender)
{
  ThumbNail1->TitlePlacement = static_cast<TTitlePos>(titlePlaceGr->ItemIndex); 
}
//---------------------------------------------------------------------------

void __fastcall TJvThumbnailChildForm::Panel6Resize(TObject *Sender)
{
  DriveComboBox2->Height = Panel9->ClientHeight;
  DriveComboBox2->Width  = Panel9->ClientWidth;
}
//---------------------------------------------------------------------------

void __fastcall TJvThumbnailChildForm::Panel8Resize(TObject *Sender)
{
  RedBar->Width = Panel8->ClientWidth;        
}
//---------------------------------------------------------------------------

void __fastcall TJvThumbnailChildForm::Panel10Resize(TObject *Sender)
{
  FilterComboBox1->Width  = Panel10->ClientWidth;
  FilterComboBox1->Height = Panel10->ClientHeight;
}
//---------------------------------------------------------------------------

File: examples/JvThumbnail/JvThumbnailChildFormU.pas
{******************************************************************

                       JEDI-VCL Demo

 Copyright (C) 2002 Project JEDI

 Original author:

 Contributor(s):

 You may retrieve the latest version of this file at the JEDI-JVCL
 home page, located at http://jvcl.sourceforge.net

 The contents of this file are used with permission, subject to
 the Mozilla Public License Version 1.1 (the "License"); you may
 not use this file except in compliance with the License. You may
 obtain a copy of the License at
 http://www.mozilla.org/MPL/MPL-1_1Final.html

 Software distributed under the License is distributed on an
 "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
 implied. See the License for the specific language governing
 rights and limitations under the License.

******************************************************************}

unit JvThumbnailChildFormU;

interface

{$I jvcl.inc}
{$I windowsonly.inc}

uses
  Classes, Controls, Forms, StdCtrls, ExtCtrls, FileCtrl, ComCtrls, JvThumbImage,
  JvThumbNails, JvBaseThumbnail, JvExExtCtrls;

type
  TJvThumbnailChildForm = class(TForm)
    Splitter2: TSplitter;
    Panel6: TPanel;
    Splitter4: TSplitter;
    DirectoryListBox2: TDirectoryListBox;
    FileListBox1: TFileListBox;
    Panel8: TPanel;
    Label6: TLabel;
    Label7: TLabel;
    Label8: TLabel;
    Label9: TLabel;
    REDBar: TTrackBar;
    GreenBar: TTrackBar;
    BlueBar: TTrackBar;
    contrastBar: TTrackBar;
    Button2: TButton;
    Panel9: TPanel;
    DriveComboBox2: TDriveComboBox;
    Panel10: TPanel;
    FilterComboBox1: TFilterComboBox;
    Panel7: TPanel;
    Panel5: TPanel;
    Label5: TLabel;
    Bevel1: TBevel;
    asbuttoncb: TCheckBox;
    autoloadcb: TCheckBox;
    minmemcb: TCheckBox;
    titleplacegr: TRadioGroup;
    Edit1: TEdit;
    GroupBox1: TGroupBox;
    Button4: TButton;
    Button5: TButton;
    Label1: TLabel;
    LightnessBar: TTrackBar;
    Button1: TButton;
    AngleGr: TRadioGroup;
    ThumbNail1: TJVThumbNail;
    ThumbImage1: TJvThumbImage;
    procedure Button2Click(Sender: TObject);
    procedure Panel6Resize(Sender: TObject);
    procedure FileListBox1Change(Sender: TObject);
    procedure asbuttoncbClick(Sender: TObject);
    procedure autoloadcbClick(Sender: TObject);
    procedure minmemcbClick(Sender: TObject);
    procedure titleplacegrClick(Sender: TObject);
    procedure Panel8Resize(Sender: TObject);
    procedure Button4Click(Sender: TObject);
    procedure Button5Click(Sender: TObject);
    procedure thumbnail1Click(Sender: TObject);
    procedure Panel10Resize(Sender: TObject);
    procedure FormShow(Sender: TObject);
    procedure AngleGrClick(Sender: TObject);
  end;

var
  JvThumbnailChildForm: TJvThumbnailChildForm;

implementation

{$R *.DFM}

procedure TJvThumbnailChildForm.Button2Click(Sender: TObject);
begin
  thumbimage1.ChangeRGB(redbar.Position,greenbar.Position,bluebar.Position);
  thumbimage1.Contrast(contrastbar.Position);
  THumbImage1.Lightness(LightnessBar.Position);
  redbar.Position:=0;
  greenbar.Position :=0;
  bluebar.Position:=0;
  contrastbar.position:=0;
  lightnessbar.Position:=0;
end;

procedure TJvThumbnailChildForm.Panel6Resize(Sender: TObject);
begin
  DriveComboBox2.Height:= panel9.ClientHeight;
  DriveComboBox2.Width := panel9.ClientWidth;
end;

procedure TJvThumbnailChildForm.FileListBox1Change(Sender: TObject);
begin
  if filelistbox1.FileName<>'' then
    thumbnail1.FileName := filelistbox1.FileName;
end;

procedure TJvThumbnailChildForm.asbuttoncbClick(Sender: TObject);
begin
  THumbnail1.Asbutton := asbuttoncb.Checked;
end;

procedure TJvThumbnailChildForm.autoloadcbClick(Sender: TObject);
begin
  thumbnail1.autoload := autoloadcb.Checked;
end;

procedure TJvThumbnailChildForm.minmemcbClick(Sender: TObject);
begin
  thumbnail1.minimizememory:=minmemcb.Checked;
end;

procedure TJvThumbnailChildForm.titleplacegrClick(Sender: TObject);
begin
  thumbnail1.TitlePlacement := ttitlepos(titleplacegr.ItemIndex);
end;

procedure TJvThumbnailChildForm.Panel8Resize(Sender: TObject);
begin
  redbar.Width := panel8.ClientWidth;
end;

procedure TJvThumbnailChildForm.Button4Click(Sender: TObject);
begin
  thumbimage1.invert;
end;

procedure TJvThumbnailChildForm.Button5Click(Sender: TObject);
begin
  thumbimage1.grayscale;
end;

procedure TJvThumbnailChildForm.thumbnail1Click(Sender: TObject);
begin
  if thumbnail1.FileName<>'' then
    thumbimage1.Loadfromfile(thumbnail1.FileName);
end;

procedure TJvThumbnailChildForm.Panel10Resize(Sender: TObject);
begin
  filtercombobox1.Width := panel10.ClientWidth;
  filtercombobox1.Height:= panel10.ClientHeight;
end;

procedure TJvThumbnailChildForm.FormShow(Sender: TObject);
begin
  //thumbimage1.Picture.Free;
  TitlePlacegr.ItemIndex := integer(thumbnail1.titlePlacement);
  Anglegr.ItemIndex := integer(thumbimage1.angle);
end;

procedure TJvThumbnailChildForm.AngleGrClick(Sender: TObject);
begin
  thumbimage1.angle := TAngle(anglegr.ItemIndex)
end;

end.
